resources/monitor_diagnostic_settings.go

```diff
@@ -13,11 +13,12 @@ import (
 
 func MonitorDiagnosticSettings() *schema.Table {
 	return &schema.Table{
-		Name:        "azure_monitor_diagnostic_settings",
-		Description: "DiagnosticSettingsResource the diagnostic setting resource",
-		Resolver:    fetchMonitorDiagnosticSettings,
-		Multiplex:   client.SubscriptionMultiplex,
-		Options:     schema.TableCreationOptions{PrimaryKeys: []string{"subscription_id", "id"}},
+		Name:         "azure_monitor_diagnostic_settings",
+		Description:  "DiagnosticSettingsResource the diagnostic setting resource",
+		Resolver:     fetchMonitorDiagnosticSettings,
+		Multiplex:    client.SubscriptionMultiplex,
+		DeleteFilter: client.DeleteSubscriptionFilter,
```

Comment:
I would just double check all resources that they have a deletefilter defined.

Reply:
Good point. I have checked all resources and found a few more without a DeleteFilter. Thanks.